plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/FeatureDevSessionContext.kt

```diff
@@ -78,6 +270,16 @@ class FeatureDevSessionContext(val project: Project, val maxProjectSizeBytes: Lo
         ignorePatternsWithGitIgnore = (ignorePatterns + parseGitIgnore().map { Regex(it) }).toList()
     }
 
+    fun isFileExtensionAllowed(file: File): Boolean {
+        // if it is a directory, it is allowed
+        if (file.isDirectory) return true
+
+        // TODO: Follow up with telemetry events to record excluded files extensions
```

Comment:
just curious about the TODO: there is any blocker for we do right now?

Reply:
I believe telemetry definition has to be defined first then can be emitted, the definition is not yet defined